sunpy/image/coalignment.py

```diff
@@ -287,6 +296,25 @@ def parabolic_turning_point(y):
     return numerator / denominator
 
 
+def check_for_nonfinite_entries(layer_image, template_image):
+    """
+    Issue a warning if there is any nonfinite entry in the layer or template images.
+
+    Parameters
+    ----------
+    layer_image : `numpy.ndarray`
+        A two-dimensional `numpy.ndarray`.
+    template_image : `numpy.ndarray`
+        A two-dimensional `numpy.ndarray`.
+    """
+    if not np.all(np.isfinite(layer_image)):
+        warnings.warn('The layer image has noninfinite entries.', SunpyUserWarning)
```

Comment:
I would like these warnings to be a little clearer, specifically I think it should answer the following two questions: What happens if I ignore this warning? What should I do to make it not occur?

Reply:
:+1: Done!